sdk/wallet-adapter/wallet-kit/src/ConnectButton.tsx

```diff
@@ -8,13 +8,23 @@ import { Button } from "./utils/ui";
 import { AccountModal } from "./AccountModal";
 import { useWalletKit } from "./WalletKitContext";
 import { formatAddress } from "@mysten/sui.js";
+import { WalletAdapter } from "@mysten/wallet-adapter-base/src/index";
 
 interface ConnectButtonProps extends ComponentProps<typeof Button> {
   connectText?: ReactNode;
+  connectedText?: string;
+  connectedCallback?: (
```

Comment:
Same here, what's the purpose of this instead of just hooks-driven interaction?

Reply:
When people use `ConnectButton` , the page interaction involved has the behavior of connecting and disconnecting. This behavior may be simple and one-time. It is actively triggered by the state change without using hooks, reducing the performance caused by repeated rendering. The probability of occurrence of the problem, so I think callback can be used here to achieve

Reply:
These components subscribe to the state changes through hooks anyway, so the performance should be effectively the same. I think we should remove these for now until we have a more compelling use-case and just encourage usage of hooks and effects.

Reply:
ok, I updated this component.